edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -368,12 +358,10 @@ async Task<bool> DeleteMessageCallback(long offset, MessageRef messageRef)
                                 {
                                     cleanupCount++;
                                 }
-
-                                messageCount = tailOffset - headOffset + 1;
                             }
 
                             // update Metrics for message counts
-                            Checkpointer.Metrics.QueueLength.Set(messageCount, new[] { endpointId, priority.ToString(), bool.TrueString });
+                            Checkpointer.Metrics.SetQueueLength(await sequentialStore.Count(), endpointId, priority.ToString());
```

Comment:
We still are setting queue length metric in two different places... If we think we can consolidate into one place, I will make that change as well.